### PR TITLE
Crop Yield Africa use case - initial dataset and example tabular regression

### LINKED ISSUE #51: Rename eo_encoders to geo_encoders
From @robknapen ; rename `eo_encoders` to `geo_encoders` for consistency with tabular data.

### PR DESCRIPTION
**Pull request by robknapen**
This Pull Request introduces the necessary data structures, configurations, and model components to support crop yield regression for East and Southern Africa.

Changes:
 - Added YieldAfricaDataset in src/data/yield_africa_dataset.py, which supports filtering by country and year, and handles both coordinate and tabular features (soil, climate, etc.).
 - Added make_model_ready_yield_africa.py script to transform raw Excel use case data into a model-ready CSV format.

Model Enhancements:
 - Refactored PredictiveModel to correctly return the loss in _step and renamed internal variables for clarity (e.g., feats to preds when referring to model output).
 - Removed hardcoded F.normalize from the forward pass of PredictiveModel to allow for more flexible encoder/head combinations.
 - Added HuberLoss as a new loss function component.

Configurations:
 - Added Hydra configurations for the Africa yield dataset, regression metrics, and experiment setup (yield_africa_tabular_reg.yaml).

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -10,6 +10,10 @@ TRAINER_PROFILE="gpu" # cpu/gpu/mps/ddp
 HF_HOME="${PROJECT_ROOT}/.cache/huggingface/" # set or will default to './.cache/huggingface/'
 DATA_DIR="${PROJECT_ROOT}/data/"  # set to your local data folder (for aether), or will default to '${PROJECT_ROOT}/data/'
 
+# When using (and downloading) TESSERA embeddings (e.g., crop yield use case)
+# Note that this folder can get large ...
+TESSERA_EMBEDDINGS_DIR="${PROJECT_ROOT}/data/cache/tessera/"
```
Comment (gabrieletijunaityte) [inline .env.example:17]: There is already a cache parameter that could be used?

Comment (robknapen) [reply .env.example:17]: Yes, potentially. I was trying to separate the tessera raw tile files (global..*)  that get downloaded and take lots of storage space from the rest. I can fix it later.

File: src/models/predictive_model.py [hunk 2]
```diff
@@ -59,38 +58,39 @@ def setup_encoders_adapters(self):
...
+            input_dim=self.geo_encoder.output_dim, output_dim=self.num_classes
         )
         self.prediction_head.configure_nn()
         if "prediction_head" not in self.trainable_modules:
             self.trainable_modules.append("prediction_head")
 
     @override
     def forward(self, batch: Dict[str, torch.Tensor]) -> torch.Tensor:
-        feats = self.eo_encoder(batch)
-        feats = F.normalize(feats, dim=-1)
```
Comment (gabrieletijunaityte) [inline src/models/predictive_model.py:78]: Is this a requirement of the loss you are using? I am pretty sure this loss implementation here was intentional (@vdplasthijs)

Comment (vdplasthijs) [reply src/models/predictive_model.py:78]: Thanks for flagging it, yes this is what we discussed (also with @robknapen ) some time ago right?
In the alignment model, we're optimising and testing the alignment of the L2-normalised embeddings, so to me it also makes sense to use L2-normalised embeddings in the prediction model (for consistency). 
We also tested this (with the geoclip encoder for butterflies I think) and L2-normalised embeddings accelerated the convergence of the MLP pred head.

Comment (robknapen) [reply src/models/predictive_model.py:78]: No because of the regression use case(s). I think we need to make the L2 normalization conditional or leave it to the geo_encoder or prediction head implementation. We probably want it for contrastive learning, but not for MLP regression.

Comment (vdplasthijs) [reply src/models/predictive_model.py:78]: Hi Rob, okay in that case it should be optional (as an arg for the prediction model). (We took it out of geo encoders and instead put it in the CL loss function, but because it also improves learning for prediction model we also included in the `forward()` here, but that indeed can be optional). Would you mind implementing that instead of the removal? Thanks!!

Comment (robknapen) [reply src/models/predictive_model.py:78]: Sure, I think making it optional is the best indeed. Should it be default on or off?

Comment (gabrieletijunaityte) [reply src/models/predictive_model.py:78]: I think default "true" (on)

**Review by gabrieletijunaityte — APPROVED** [review]

**Comment by robknapen** [comment]
Thanks for merging it @gabrieletijunaityte !
